File: kotlin/src/main/kotlin/io/github/ocirne/cryptopals/set1/Challenge1.kt
package io.github.ocirne.cryptopals.set1

import io.github.ocirne.cryptopals.Basics

class Challenge1 {

    fun run(hexString: String): String {
        val byteArray = Basics.decodeHexString(hexString)
        return Basics.encodeToBase64(byteArray)
    }
}

File: python/src/cryptopals/set1/challenge1.py
import base64
from cryptopals.basics import from_hex


def hex_to_base64(hex_string):
    """
    >>> hex_to_base64('49276d206b696c6c696e6720796f757220627261696e206c696b65206120706f69736f6e6f7573206d757368726f6f6d')
    b'SSdtIGtpbGxpbmcgeW91ciBicmFpbiBsaWtlIGEgcG9pc29ub3VzIG11c2hyb29t'
    """
    byte_array = from_hex(hex_string)
    return base64.b64encode(byte_array)
